conf/config.go

```diff
@@ -122,40 +122,49 @@ func LoadConfig(mainConfigFile string, fallbackConfigFile string) (*MenderConfig
 		return nil, loadErr
 	}
 
+	log.Debugf("Loaded %d configuration file(s)", filesLoadedCount)
+
 	if filesLoadedCount == 0 {
 		log.Info("No configuration files present. Using defaults")
 		return config, nil
 	}
 
-	if config.Servers == nil {
-		if config.ServerURL == "" {
+	log.Debugf("Loaded configuration = %#v", config)
+
+	return config, nil
+}
+
+// VerifyServerConfig verifies the Servers fields in the configuration
+func (c *MenderConfig) VerifyServerConfig() error {
+	if c.Servers == nil {
+		if c.ServerURL == "" {
 			log.Warn("No server URL(s) specified in mender configuration.")
 		}
-		config.Servers = make([]client.MenderServer, 1)
-		config.Servers[0].ServerURL = config.ServerURL
-	} else if config.ServerURL != "" {
+		c.Servers = make([]client.MenderServer, 1)
+		c.Servers[0].ServerURL = c.ServerURL
```

Comment:
You may also want to move this part to loadConfig. The validation function shouldn't modify the contents but report if their wrong IMO.

Reply:
Well, that defeats the purpose of doing this work, as the warning would be printed at `Load` time...

Reply:
Ahh, you're right. Ignore what I said, I suppose it'll require substantial refactoring to achieve what I was suggesting.